File: GoF_Design_Pattern/Adapter_Pattern/Adapter_with_Delegation/java/PrintBanner.java
package Software_Engineering.GoF_Design_Pattern.Adapter_Pattern.Adapter_with_Delegation.java;


public class PrintBanner extends Print {

    private Banner banner;

    public PrintBanner(String string) {
        banner = new Banner(string);
    }

    @Override
    public void printWeak() {
        banner.showWithParen();
    }

    @Override
    public void printStrong() {
        banner.showWithAster();
    }

}

File: GoF_Design_Pattern/Adapter_Pattern/Adapter_with_Delegation/python/PrintBanner.py
#! /etc/anaconda/python3

from Print import Print
from Banner import Banner


class PrintBanner(Print):

    __banner = None

    # def __new__(cls, *args, **kwargs):
    #     return Print()

    def __init__(self, string):
        self.__banner = Banner(string)

    def print_weak(self):
        self.__banner.show_with_paten()

    def print_strong(self):
        self.__banner.show_with_aster()
